vue/components/ui/molecules/carousel/carousel.vue

```diff
@@ -0,0 +1,307 @@
+    transition: scale ease-in-out 0.1s, transform ease-in-out 0.1s;
+    z-index: 1;
+}
+
+.carousel .arrow:hover {
+    scale: 1.3;
+}
+
+.carousel .arrow:active {
+    scale: 1.4;
+}
+
+.carousel .arrow-previous {
+    left: 0px;
+}
+
+.carousel .arrow-next {
+    right: 0px;
+}
+
+.carousel .arrow-previous:active {
+    transform: translate(-5px);
+}
+
+.carousel .arrow-next:active {
+    transform: translate(5px);
+}
+
+.carousel .slide-container {
+    height: 100%;
+    position: relative;
+    text-align: left;
+    white-space: nowrap;
+    width: 100%;
+}
+
+.carousel .slide-container .slide {
+    cursor: pointer;
+    display: inline-block;
+    height: 100%;
+    position: absolute;
+    text-align: center;
+    vertical-align: top;
+    width: 100%;
+}
+
+.carousel .slide-container .slide .slide-content {
+    align-items: center;
+    display: flex;
+    height: 100%;
+    justify-content: center;
+    width: 100%;
+}
+
+.carousel .slide-container .slide .slide-content .image {
+    height: 100%;
+    pointer-events: none;
+    width: 100%;
+}
+
+.fade-enter-active,
+.fade-leave-active {
+    transition: opacity 0.5s;
+}
+
+.fade-enter,
+.fade-leave-to {
+    opacity: 0;
+}
+</style>
+
+<script>
+export const Carousel = {
+    name: "carousel",
+    props: {
+        /**
+         * Set of items to be displayed in the carousel slider.
+         */
+        items: {
+            type: Array,
+            default: () => []
+        },
+        /**
+         * The index of the image to be displayed first.
+         */
+        value: {
+            type: Number,
+            default: 0
+        },
+        /**
+         * The width of the carousel in pixels.
+         */
+        width: {
+            type: Number,
+            default: 320
+        },
+        /**
+         * The height of the carousel in pixels.
+         */
+        height: {
+            type: Number,
+            default: 400
+        },
+        /**
+         * Weather or not the arrows to navigate between slides should show.
+         */
+        arrows: {
+            type: Boolean,
+            default: true
+        },
+        /**
+         * The size of the arrows to move between the slide items in pixels.
+         */
+        arrowsSize: {
+            type: Number,
+            default: 36
+        },
+        /**
+         * The color of the arrows to move between the slide items.
+         */
+        arrowsColor: {
+            type: String,
+            default: "#808080"
+        },
+        /**
+         * The fit style of the images.
+         */
+        imageFit: {
+            type: String,
+            default: "contain"
+        },
+        /**
+         * The amount of pixels that need to be dragged to trigger a swipe.
+         */
+        swipeThreshhold: {
+            type: Number,
+            default: 50
+        }
+    },
+    data: function() {
+        return {
+            dragStartPosition: null,
+            loadedItems: 0,

```

Comment:
@3rdvision I'm removing this feature for now because, while it's a good idea, whenever we tackle the loading problem for this component we'll have to do more than this:
- Fine-grained loading events: knowing when each image loaded individually, instead of using a global percentage
- Individual or global loading indicator(s)

Reply:
.. and tests :p